Plugins marketplace: Change formatting of search and filtering

xavier-lc opened the pull request:
Related to https://github.com/Automattic/dotcom-forge/issues/7537

## Proposed Changes

* Update the design of the search and filters on `/plugins` to the one proposed on the linked issue.
* Figma: https://www.figma.com/design/kyCC1EHKtv4qsElmlAzp5T/Flyout-panel-i2?node-id=537-123318&m=dev

## Testing Instructions

* Open the live preview and go to `/plugins`
* Check that the search and filters still work, and look like the proposed design

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Bring changes from `update/plugins-browser`

Commit:
Add outline to the focused element

xavier-lc commented on client/my-sites/plugins/plugins-browser/index.jsx:
```diff
@@ -171,33 +167,15 @@ const PluginsBrowser = ( { trackPageViews = true, category, search, hideHeader }
+					searchRef={ searchRef }
 					searchTerm={ search }
-					isSearching={ isFetchingPluginsBySearchTerm }
-					title={
-						'en' === locale || hasTranslation( 'Flex your site’s features with plugins' )
-							? __( 'Flex your site’s features with plugins' )
-							: __( 'Plugins you need to get your projects done' )
-					}
-					subtitle={
-						! isLoggedIn &&
-						( 'en' === locale ||
-							hasTranslation(
-								'Add new functionality and integrations to your site with thousands of plugins.'
-							) ) &&
-						__( 'Add new functionality and integrations to your site with thousands of plugins.' )
-					}
```
Title and subtitle are no longer needed.

Commit:
Move `useTermsSuggestions` to `search-categories` folder

Commit:
Remove `search-box-header` folder

The relevant code had already been moved to `search-categories`

xavier-lc commented on client/my-sites/plugins/search-categories/index.tsx:
```diff
@@ -0,0 +1,166 @@
+
+import './style.scss';
+
+const SearchBox: FC< {
```
Component taken from the `search-box-header` folder.

mmtr replied:
What's preventing you using from the existing component?

xavier-lc commented on client/my-sites/plugins/search-categories/index.tsx:
```diff
@@ -0,0 +1,166 @@
+	);
+};
+
+const SearchCategories: FC< {
```
The search logic has also been taken from the `search-box-header` folder.

mmtr replied:
Same as above, what's preventing you from using the existing logic?

Commit:
Remove unnecessary `isMobile` logic

Given that it's always set to `false`, we don't need it

BogdanUngureanu commented:
The design looks great on desktop resolutions. However, on mobile, it looks a bit off to me. 
![Screenshot 2024-06-07 at 11 42 41](https://github.com/Automattic/wp-calypso/assets/995955/9683cc70-08a8-46bf-887b-6b9771b60661)

Is that expected? I couldn't find a mockup for mobile resolutions in the Figma file. 

Maybe we want to increase the size of the inputs?

mmtr commented:
This will also change the logged-out plugins marketplace:

Before | After
--- | ---
<img width="1392" alt="Screenshot 2024-06-07 at 14 09 36" src="https://github.com/Automattic/wp-calypso/assets/1233880/ab59d7a0-da26-4456-8b12-766a7d900504"> | <img width="1392" alt="Screenshot 2024-06-07 at 14 09 39" src="https://github.com/Automattic/wp-calypso/assets/1233880/74b3c4ac-0a6f-4477-b82f-a8b19cb95633">

Is that intentional? Looking at the issue, it seems we want to target the design to the logged-in page only.

xavier-lc commented:
> Is that expected? I couldn't find a mockup for mobile resolutions in the Figma file.

There is none 😅

> Maybe we want to increase the size of the inputs?

Let me try 👍

Commit:
Adjust mobile style

xavier-lc commented:
@BogdanUngureanu I've widened the input and adjusted the size of the button.

BogdanUngureanu commented:
The changes look great on mobile now, thanks! The only thing that remains is the logged-out view and we can ship it!

Commit:
Revert "Remove `search-box-header` folder"

This reverts commit 300a7ce9f353248624b285a58cdc365248867303.

Commit:
Move `useTermsSuggestions` to its own folder

Commit:
Restore logged out look

Commit:
Fix style change

Commit:
Remove logged out style, which will be handled by another component

xavier-lc commented:
@BogdanUngureanu the logged out page is back to how it was.

mmtr commented:
Found a few differences when comparing with production.

While logged-out, the top navigation bar was blue, it is now black:

Before | After
--- | ---
<img width="1283" alt="Screenshot 2024-06-11 at 13 09 05" src="https://github.com/Automattic/wp-calypso/assets/1233880/c05a744c-909a-4479-a85a-d4ff06373488"> | <img width="1278" alt="Screenshot 2024-06-11 at 13 09 13" src="https://github.com/Automattic/wp-calypso/assets/1233880/4289d632-ee02-4e24-89ab-1189db9122bb">


While logged in, if you reload the page, there is a brief moment where the logged-out page shows up which does not happen in prod:

**Before**

https://github.com/Automattic/wp-calypso/assets/1233880/b0638e4b-80fc-48f1-b747-36d573bde221

**After**


https://github.com/Automattic/wp-calypso/assets/1233880/a5283718-363b-43a9-8662-8bee10097e50

mmtr commented:
On mobile viewports, it's a bit weird to have a dropdown that says "More" with no additional context:

<img width="325" alt="Screenshot 2024-06-11 at 13 15 33" src="https://github.com/Automattic/wp-calypso/assets/1233880/dc954d4a-5997-4f74-8d96-1da5ed365f30">

It could maybe display the current selected option.

mmtr commented:
This is also breaking a few E2E tests:

> specs/plugins/plugins__browse.ts: Plugins: Browse: Can browse SEO category
> specs/plugins/plugins__search.ts: Plugins Search: Navigate back to the default plugins page when searching from categories pages

xavier-lc commented:
> While logged-out, the top navigation bar was blue, it is now black:

Fixed after rebase.

> While logged in, if you reload the page, there is a brief moment where the logged-out page shows up which does not happen in prod:

I think that's got to do with the dev environment. I see the same behavior on `trunk`, and I've also checked the live preview for 2 unrelated PRs and the same happens there.

mmtr commented:
> I think that's got to do with the dev environment. I see the same behavior on trunk, and I've also checked the live preview for 2 unrelated PRs and the same happens there.

Thanks for confirming!

The only remaining blockers are the E2E failures.

The one about the dropdown on mobile can be addressed on a follow-up.

xavier-lc commented:
> On mobile viewports, it's a bit weird to have a dropdown that says "More" with no additional context:

Yeah, it's how the `DropdownGroup` component [works](https://github.com/Automattic/wp-calypso/blob/trunk/packages/components/src/responsive-toolbar-group/dropdown-group.tsx#L108). I could leave it as it currently works with a swiper below the search field.

Commit:
Fix e2e tests

xavier-lc commented:
> The only remaining blockers are the E2E failures.

Fixed now 🥳️

taipeicoder commented:
Jotting down some minor UI issues in the logged-in `/plugins/:site`

Subtle border.
![Screenshot 2024-06-12 at 2 35 29 PM](https://github.com/Automattic/wp-calypso/assets/797888/8ef17731-45e2-4c01-adf1-d0c1d8eaf91e)

Outline color in selected category.
![Screenshot 2024-06-12 at 12 18 19 PM](https://github.com/Automattic/wp-calypso/assets/797888/14a95599-9a8f-4599-a6c0-ee6095df6505)

Odd categories that are not present in production?
![Screenshot 2024-06-12 at 12 19 20 PM](https://github.com/Automattic/wp-calypso/assets/797888/377f0329-3b48-4b32-8449-c90ee55b9ea1)

Commit:
PR feedback, remove border

Commit:
PR feedback, remove outline when tab is active

xavier-lc commented:
> Subtle border.

Good catch, fixed on https://github.com/Automattic/wp-calypso/pull/91565/commits/00040f6085cee8ac554ebe9259b43e62100ca91f

> Outline color in selected category.

Good catch, fixed on https://github.com/Automattic/wp-calypso/pull/91565/commits/5815e2c898d727011a2f02f50fe5db8cd814ea9c

> Odd categories that are not present in production?

The logic that handles the categories hasn't been changed. I'm not sure why that category is not on the dropdown, but it does show up on the main page:

![image](https://github.com/Automattic/wp-calypso/assets/8511199/13be832a-c9d5-4292-a3c2-c33cef0c3b31)

You can see it on our codebase as well:

https://github.com/Automattic/wp-calypso/blob/f65ce82fa09fc78f424571c25f900b550d471445/client/my-sites/plugins/categories/use-categories.tsx#L332-L334

> Container width jumping around when switching from Discover to another category:

This has been discussed with @lucasmendes-design (see https://github.com/Automattic/dotcom-forge/issues/7491#issuecomment-2155360441) and will be addressed on a followup PR.

mmtr commented:
> The logic that handles the categories hasn't been changed. I'm not sure why that category is not on the dropdown, but it does show up on the main page:

Presumably, we're hiding the categories that are displayed as collections in the discover page. We should keep that behavior

mmtr commented:
> Presumably, we're hiding the categories that are displayed as collections in the discover page. We should keep that behavior

I guess the fact that the behavior is now accidentally different, is a side effect of duplicating some logic into the new `search-categories` folder. Can we re-use the existing logic instead of duplicating it?

taipeicoder commented:
> Presumably, we're hiding the categories that are displayed as collections in the discover page. We should keep that behavior

+1. It's odd that we show them in the desktop viewport:

![Screenshot 2024-06-12 at 9 43 03 PM](https://github.com/Automattic/wp-calypso/assets/797888/8840d3f6-7f7b-4079-a248-e7578dd6e3bb)

But in the mobile viewport we show the dropdown, which doesn't have those categories:

![Screenshot 2024-06-12 at 9 42 57 PM](https://github.com/Automattic/wp-calypso/assets/797888/fc43e734-51ce-4280-8478-40fc54342ba6)

Commit:
Add `titleField` on `ScrollableHorizontalNavigation` so the proper menu title is shown for categories

xavier-lc commented:
Found it, I needed to use the `menu` key from the `Category` objects instead of the `title` one. Fixed on https://github.com/Automattic/wp-calypso/pull/91565/commits/d6177baac609a9c723190abd14b51c8056920e23

taipeicoder reviewed (approved):
Thanks for working on this, the changes LGTM 👍
